openff/toolkit/tests/test_parameters.py

```diff
@@ -2188,6 +2181,24 @@ def generate_v_site_coordinates(
 
         return output_conformer[molecule.n_atoms :, :]
 
+    def test_serialize_roundtrip(self):
+
+        force_field = ForceField()
+
+        handler = force_field.get_parameter_handler("VirtualSites")
+        handler.add_parameter(
+            parameter=VirtualSiteMocking.bond_charge_parameter("[*:1][*:2]")
+        )
+        handler.add_parameter(
+            parameter=VirtualSiteMocking.monovalent_parameter("[*:1][*:2][*:3]")
+        )
+        handler.add_parameter(
+            parameter=VirtualSiteMocking.divalent_parameter("[*:2][*:1][*:3]", "once")
+        )
+        handler.add_parameter(
+            parameter=VirtualSiteMocking.trivalent_parameter("[*:1][*:2][*:3][*:4]")
+        )
+
     @pytest.mark.parametrize(
         "smiles, matched_indices, parameter, expected_raises",
```

Comment:
Note to self: I think this test is either misnamed or incomplete, I will fix later